crates/ra_syntax/src/ast/make.rs

```diff
@@ -110,6 +110,22 @@ pub fn match_arm_list(arms: impl Iterator<Item = ast::MatchArm>) -> ast::MatchAr
     }
 }
 
+pub fn let_match_early(expr: ast::Expr, early_expression: &str) -> ast::LetStmt {
```

Comment:
This is a bit too high-level code for `make`. Ideally, `make` should provides only simple "constructor" functions for primitive expressions, and everything more complex should be assembled on top. I am not sure what's the best way to do this in this particular case, I can take a closer look after this PR is merged. 

Ideally, we'd want quasi-quoting a-la `quote!`, but I belie there's a long way until that...

Reply:
I left this one as is and addresses your other comments.